app/src/main/java/com/swent/assos/config/Config.kt

```diff
@@ -1,10 +1,78 @@
 package com.swent.assos.config
 
+import java.net.HttpURLConnection
+import java.net.URL
+import org.json.JSONObject
+
 class Config {
+  data class Service(val name: String, val host: String, val port: Int?)
+
+  fun checkFirebaseEmulator(ip: String, port: String): Boolean {
+    val url = URL("http://${ip}:${port}")
+    val connection = url.openConnection() as HttpURLConnection
+    connection.requestMethod = "GET"
+
+    var connected: Boolean = false
+    var responseCode: Int
+    val thread = Thread {
+      try {
+        connection.connect()
+        responseCode = connection.responseCode
+
+        if (responseCode == HttpURLConnection.HTTP_OK) {
+          println("Firebase Authentication Emulator is online.")
+          connected = true
+        } else {
+          println(
+              "Firebase Authentication Emulator is not yet online. Response code: $responseCode")
+        }
+      } catch (e: Exception) {
+
+        println("Error checking Firebase Authentication Emulator status: ${e.message}")
+      } finally {
+        connection.disconnect()
+      }
+    }
+    thread.start()
+    thread.join()
```

Comment:
Code smells, there is surely a better way to deal with this thread (just run a code rather than wrap it in a thread)